dataactvalidator/scripts/load_program_activity.py

```diff
@@ -20,54 +29,179 @@ def get_program_activity_file(base_path):
+    if not last_stored_obj:
+        # return epoch ts to make sure we load the data the first time through,
+        # and ideally any time the data might have been wiped
+        last_stored = datetime.datetime.utcfromtimestamp(0)
+    else:
+        last_stored = last_stored_obj.last_load_date
+    return last_stored
+
+
+def set_stored_pa_last_upload(load_datetime):
+    """ Set upload date of most recent file we have recorded (Datetime object)
+
+        Args:
+            Datetime object representing the timestamp associated with the current file
+    """
+    sess = GlobalDB.db().session
+    last_stored_obj = sess.query(ExternalDataLoadDate).join(ExternalDataType).filter(
+        ExternalDataType.name == "program_activity_upload"
+        ).one_or_none()
+    if not last_stored_obj:
+        data_type = sess.query(ExternalDataType).filter(
+            ExternalDataType.name == "program_activity_upload"
+            ).one_or_none()
+        last_stored_obj = ExternalDataLoadDate(external_data_type_id=data_type.external_data_type_id,
+                                               last_load_date=load_datetime)
+        sess.add(last_stored_obj)
+    else:
+        last_stored_obj.last_load_date = load_datetime
+    sess.commit()
+
+
 def load_program_activity_data(base_path):
     """ Load program activity lookup table.
 
         Args:
             base_path: directory of domain config files
     """
+    last_upload = get_date_of_current_pa_upload(base_path)
+    if not (last_upload > get_stored_pa_last_upload()):
+        return
+    else:
+        set_stored_pa_last_upload(last_upload)
 
     program_activity_file = get_program_activity_file(base_path)
 
-    logger.info('Loading program activity: program_activity.csv')
+    logger.info('Loading program activity: ' + PA_FILE_NAME)
 
     with create_app().app_context():
         sess = GlobalDB.db().session
+        try:
+            data = pd.read_csv(program_activity_file, dtype=str)
+        except pd.io.common.EmptyDataError as e:
+            log_blank_file()
+            sys.exit(4)  # exit code chosen arbitrarily, to indicate distinct failure states. Approved by Ops.
+        try:
+            dropped_count, data = clean_data(
+                data,
+                ProgramActivity,
+                {"fyq": "fiscal_year_quarter", "agency_id": "agency_id", "allocation_id": "allocation_transfer_id",
+                 "account": "account_number", "pa_code": "program_activity_code", "pa_name": "program_activity_name"},
+                {"program_activity_code": {"pad_to_length": 4}, "agency_id": {"pad_to_length": 3},
+                 "allocation_transfer_id": {"pad_to_length": 3, "keep_null": True},
+                 "account_number": {"pad_to_length": 4}},
+                ["agency_id", "program_activity_code", "account_number", "program_activity_name"],
+                True
+            )
+        except FailureThresholdExceededException as e:
+            if e.count == 0:
```

Comment:
Can this happen considering the above exception? According to the pandas documentation, the above exception is raised when `Exception that is thrown in pd.read_csv (by both the C and Python engines) when empty data or header is encountered.` so I don't think we'd ever get a count of 0? If it can happen, then does the above exception ever happen?